pkg/httputils/client.go

```diff
@@ -0,0 +1,87 @@
+package httputils
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// ParseHost inputs a host address string, and output three type:
+// url.URL, basePath and an error.
+func ParseHost(host string) (*url.URL, string, string, error) {
```

Comment:
In the comment, you wrote that this function returns three types. While I found that this function actually returns four types `(*url.URL, string, string, error)`.
I am afraid in consistency exists.